### PR TITLE
Don't test for agent log in between tests ETP

### PR DESCRIPTION
**Pull request by larohra**
## Description
Currently we were checking for errors in agent log mid-test in ETP. Refactored that to only check at the end and check for relevant errors mid-test

Issue # 

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: dcr/scenario_utils/check_waagent_log.py [hunk 1]
```diff
@@ -93,6 +92,21 @@ def check_waagent_log_for_errors(waagent_log=AGENT_LOG_FILE, ignore=None):
         {
             'message': r"WARNING EnvHandler ExtHandler Dhcp client is not running."
         },
+        # This happens in CENTOS and RHEL when waagent attempt to format and mount the error while cloud init is already doing it
```
Comment (narrieta) [inline dcr/scenario_utils/check_waagent_log.py:95]: do we get these errors at all in DCR V2?

Comment (larohra) [reply dcr/scenario_utils/check_waagent_log.py:95]: haven't seen it honestly. I just copied the complete ignore list from v1 and copied it in v2.

Comment (narrieta) [reply dcr/scenario_utils/check_waagent_log.py:95]: we wouldn't expect it, right? sounds like a provisioning error. i'd suggest removing it here and we can always add it back if there a reason to do so

Comment (larohra) [reply dcr/scenario_utils/check_waagent_log.py:95]: makes sense, let me remove it

**Review by narrieta — APPROVED** [review]

**Review by kevinclark19a — APPROVED** [review]

**Pull request merged**